chronicles.nim

```diff
@@ -136,6 +136,24 @@ else:
 
 type InstInfo = tuple[filename: string, line: int, column: int]
 
+when compileOption("threads"):
+  # With threads turned on, we give the thread id
+  # TODO: Does this make sense on all platforms? On linux, conveniently, the
+  #       process id is the thread id of the `main` thread..
+  proc getLogThreadId*(): int = getThreadId()
+else:
+  # When there are no threads, we show the process id instead, allowing easy
+  # correlation on multiprocess systems
+  when defined(linux) or defined(macosx):
```

Comment:
This can be just `defined(posix)` to allow bsd and others into the party.

Reply:
fixed, nice